depthai.py

```diff
@@ -131,6 +147,66 @@ def stream_type(option):
         stream_dict = {'name': stream_name, "max_fps": max_fps}
     return stream_dict
 
+def decode_tracklets(packetData):
+    packetData.dtype = np.int32
+    tracklets = []
+    nr_objects = packetData[0]
+    tracklet_base = packetData[2:]
+    tracklet_size = 8
+    for i in range(nr_objects):
+        tracklet_base
+        decoded_tracklet = Tracklet()
+        tracklet_start = i * tracklet_size
+        tracklet_end = (i + 1) * tracklet_size
+        tracklet = tracklet_base[tracklet_start:tracklet_end]
+        decoded_tracklet.left = tracklet[0]
+        decoded_tracklet.top = tracklet[1]
+        decoded_tracklet.right = tracklet[2]
+        decoded_tracklet.bottom = tracklet[3]
+        decoded_tracklet.id = tracklet[4]
+        decoded_tracklet.label = tracklet[6]
+        decoded_tracklet.status = Tracklet.states[tracklet[7]]
+        tracklets.append(decoded_tracklet)
+
+    return tracklets
```

Comment:
All of your tracklet objects will have the same values :/ -- I believe this is a bug.

Reply:
This is not currently a bug, so please ignore my comment above. However it could very easily become a bug.

Reply:
Maybe just pass the `tracklet_base[tracklet_start:tracklet_end]` to __init__ function of the `Tracklet()`?

Reply:
Yeah, or another way I've seen this done is to have a `classmethod` constructor, which is pretty nice to use!

```python
class Tracklet:
    def __init__(self):
        # some stuff...
    
    @classmethod
    def from_packet(cls, packet_data):
        obj = cls()
        obj.instance_var = packet_data[#...]
        # ...
        return obj

# ... 

def decode_tracklets(packet_data):
    # ...
    tracket = Tracklet.from_packet(packet_data)
```

It's a lot to write it all out, but that's the gist.

Reply:
Nice, love it!